### PR TITLE
Fix bug in `ImageClassifcation` task template

### PR DESCRIPTION
**Pull request by mariosasko**
Fixes a bug in the `ImageClassification` task template which requires specifying class labels twice in dataset scripts. Additionally, this PR refactors the API around the classification task templates for cleaner `labels` handling.

CC: @lewtun @nateraw

### CODE REVIEW

**Comment by mariosasko** [comment]
The CI failures are unrelated to the changes in this PR.

File: src/datasets/arrow_dataset.py [hunk 1]
```diff
@@ -1867,14 +1866,9 @@ def prepare_for_task(self, task: Union[str, TaskTemplate]) -> "Dataset":
...
-                f"Expected a `str` or `datasets.tasks.TaskTemplate` object but got task {task} with type {type(task)}."
+                f"Expected a `str` or `datasets.TaskTemplate` object but got task {task} with type {type(task)}."
             )
-        if isinstance(template, TextClassification) and self.info.features is not None:
-            dataset_labels = tuple(sorted(self.info.features[template.label_column].names))
-            if template.labels is None or template.labels != dataset_labels:
-                raise ValueError(
-                    f"Incompatible labels between the dataset and task template! Expected labels {dataset_labels} but got {template.labels}. Please ensure that `datasets.tasks.TextClassification.labels` matches the features of the dataset."
-                )
+        template = template.align_with_features(self.info.features)
```
Comment (lewtun) [inline src/datasets/arrow_dataset.py:1871]: This is much more elegant - thanks for refactoring the original implementation!

**Review by lewtun — APPROVED** [review]
Thank you for this great refactoring of the task template! LGTM 🚀

**Comment by lewtun** [comment]
> The CI failures are unrelated to the changes in this PR.

It seems that some of the failures are due to the tests on the dataset cards (e.g. CIFAR, MNIST, FASHION_MNIST). Perhaps it's worth addressing those in this PR to avoid confusing downstream developers who branch off `master` and suddenly have a failing CI?

**Pull request merged**